domain/cloudimagemetadata/state/types.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import "time"
+
+// inputMetadata represents metadata information for a cloud image.
+// It is used to prepare and execute sqlair statement.
+type inputMetadata struct {
+	UUID            string    `db:"uuid"`
+	CreatedAt       time.Time `db:"created_at"`
+	Source          string    `db:"source"`
+	Stream          string    `db:"stream"`
+	Region          string    `db:"region"`
+	Version         string    `db:"version"`
+	VirtType        string    `db:"virt_type"`
+	RootStorageType string    `db:"root_storage_type"`
+	RootStorageSize *uint64   `db:"root_storage_size"`
+	Priority        int       `db:"priority"`
+	ArchitectureID  int       `db:"architecture_id"`
+	ImageID         string    `db:"image_id"`
+}
+
+// outputMetadata represents metadata information related to a cloud image.
+// It is used to prepare and execute sqlair statement.
+type outputMetadata struct {
+	CreatedAt        time.Time `db:"created_at"`
+	Source           string    `db:"source"`
+	Stream           string    `db:"stream"`
+	Region           string    `db:"region"`
+	Version          string    `db:"version"`
+	VirtType         string    `db:"virt_type"`
+	RootStorageType  string    `db:"root_storage_type"`
+	RootStorageSize  *uint64   `db:"root_storage_size"`
+	Priority         int       `db:"priority"`
+	ArchitectureName string    `db:"architecture_name"`
+	ImageID          string    `db:"image_id"`
+}
+
+// metadataImageID represents a structure holding an image ID for metadata processing in the database.
+// It is used to prepare and execute sqlair statement.
+type metadataImageID struct {
+	ID string `db:"image_id"`
+}
+
+// metadataUUID represents a structure that encapsulates a universally unique identifier (UUID) for metadata purposes.
+// It is used to prepare and execute sqlair statement.
+type metadataUUID struct {
+	UUID string `db:"uuid"`
+}
+
+type ttl struct {
```

Comment:
Don't I also deserve a comment?

Reply:
You do because I commented other non exposed symbol in this file, I'm sorry to hurt your feeling, dear `ttl` symbol.